### ISSUE TITLE
pocolog to rosbag conversion

### ISSUE THREAD

**Issue by bkocev** [rock-core/base-console_bridge#5]
Hi everyone,

I am investigating how to convert pocolog files (logs generated with Rock) to rosbags (ROS-compliant log files) in order to be able to reuse data within ROS.

Has this been addressed somehow until now or maybe could you please guide me how to solve this task?

Thank you and kind regards,
Bojan Kocev.

**Comment by doudou** [comment]
> Has this been addressed somehow until now or maybe could you please guide me how to solve this task?

I don't think it has. The main issue (I guess) is the conversion between the Rock type and the ROS message.

If I had to do it, I would look into:
- load data using pocolog_cpp
- using the rock-to-ros conversions that are part of the orogen integration for the ROS bridge to convert to a ROS message
- use ROS C++ library to write rosbag (I'm guessing that exists).
- 
Not a small amount of work, though.

**Comment by bkocev** [comment]
Hi,

thanks for the answer. 
Currently, I am trying to solve my original task using the rock-ros-bridge, which should handle the conversion from Rock types to ROS messages. Some of the things you mentioned above are already being used by the bridge as far as I know.

Thanks,
Bojan.
